### PR TITLE
liuliu/get-pull-and-update-pull-doc

### PR DESCRIPTION
**Pull request by liuliu-dev**

### CODE REVIEW

File: content/.gitbook/assets/getpull.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+          {
+            "name": "pull_id",
+            "in": "path",
+            "required": true,
+            "description": "ID of the pull request",
+             "schema": {
+              "type": "string",
+              "example":"1"
+            }
+          }
```
Comment (tbantle22) [inline content/.gitbook/assets/getpull.json]: What about `owner` and `database`?

File: content/.gitbook/assets/getpull.json [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    },
+                    "title:": {
+                      "type": "string",
+                      "description": "Title of the pull request",
+                      "example":"Added new data"
+                    },
+                    "description:": {
+                      "type": "string",
+                      "description": "Description of the pull request",
+                      "example":"Added new data"
```
Comment (tbantle22) [inline content/.gitbook/assets/getpull.json]: Maybe something different than the title like "Added missing museums, sourced from museums.com"

File: content/.gitbook/assets/getpull.json [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                    "to_branch_database":{
+                      "type": "string",
+                      "description": "Database name of the branch to which the pull request was created",
+                      "example":"museum-collections"
+                    },
+                    "to_branch_name":{
+                      "type": "string",
+                      "description": "Name of the branch to which the pull request was created",
+                      "example":"main"
+                    },
```
Comment (tbantle22) [inline content/.gitbook/assets/getpull.json]: These descriptions are a little unclear. There needs to be a difference in the descriptions for `from_x` vs `to_x`

File: content/.gitbook/assets/updatepull.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+          {
+            "name": "pull_id",
+            "in": "path",
+            "required": true,
+            "description": "ID of the pull request to update.",
+            "schema": {
+              "type": "string",
+              "example": "1"
+            }
+          }
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]: `owner` and `database` here too

File: content/.gitbook/assets/updatepull.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        ],
+        "requestBody": {
+          "description": "Request body containing the updated pull request information.",
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]: Do all of these need to be populated? Or only the fields I want to update?

File: content/.gitbook/assets/updatepull.json [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                    "example": "Added new data"
+                  },
+                  "description": {
+                    "type": "string",
+                    "description": "The updated description of the pull request.",
+                    "example": "Added new data from LACMA museum."
+                  },
+                  "state": {
+                    "type": "string",
+                    "description": "The updated state of the pull request ('closed' to close the pull request).",
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]:
```suggestion
                    "description": "The updated state of the pull request (can only update to 'closed')",
```

File: content/.gitbook/assets/updatepull.json [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+                      "example": "Added new data"
+                    },
+                    "description": {
+                      "type": "string",
+                      "description": "Description of the pull request after update",
+                      "example": "Added new data from LACMA museum."
+                    },
+                    "state": {
+                      "type": "string",
+                      "description": "State of the pull request after update ('closed' if closed)",
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]:
```suggestion
                      "description": "State of the pull request after update",
```

File: content/.gitbook/assets/updatepull.json [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+                      "example": "closed"
+                    },
+                    "allow_maintainer_to_edit_from_branch": {
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]: Can you not update this field? If not we should exclude it here

File: content/.gitbook/assets/updatepull.json [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+                      "type": "string",
+                      "description": "Whether maintainers are allowed to edit from the branch",
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]:
```suggestion
                      "description": "If the pull request is from a fork, allows database maintainers to edit the from_branch",
```

File: content/.gitbook/assets/updatepull.json [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+                "schema": {
+                  "type": "object",
+                  "properties": {
+                    "status": {
+                      "type": "string",
+                      "example": "Error"
+                    },
+                    "message": {
+                      "type": "string",
+                      "example": "Error updating a pull request."
```
Comment (tbantle22) [inline content/.gitbook/assets/updatepull.json]:
```suggestion
                      "example": "Error updating the pull request."
```

File: content/products/dolthub/api/database.md [hunk 11]
```diff
@@ -36,11 +36,43 @@ headers = {
...
 ```
 
-{% swagger src="../../../.gitbook/assets/pullrequest.json" path="/{owner}/{database}/pulls" method="post" %}
-[pullrequest.json](../../../.gitbook/assets/pullrequest.json)
+{% swagger src="../../../.gitbook/assets/createpull.json" path="/{owner}/{database}/pulls" method="post" %}
+[createpull.json](../../../.gitbook/assets/createpull.json)
 {% endswagger %}
 
 
+## Get details of a pull request
```
Comment (tbantle22) [inline content/products/dolthub/api/database.md]:
```suggestion
## Get pull request details
```

File: content/products/dolthub/api/database.md [hunk 12]
```diff
@@ -36,11 +36,43 @@ headers = {
...
+
+This API allows you to retrieve the details of a specific pull request in the `museum-collections`` database. In this example, we will retrieve the details of pull request #1.
```
Comment (tbantle22) [inline content/products/dolthub/api/database.md]:
```suggestion
This API allows you to retrieve the details of a specific pull request in the `museum-collections` database. In this example, we will retrieve the details of pull request #1.
```

File: content/products/dolthub/api/database.md [hunk 13]
```diff
@@ -36,11 +36,43 @@ headers = {
...
+}
+```
+
+{% swagger src="../../../.gitbook/assets/getpull.json" path="/{owner}/{database}/pulls/{pull_id}" method="get" %}
+[getpull.json](../../../.gitbook/assets/getpull.json)
+{% endswagger %}
+
+## Update a pull request
+
+This API allows you to update a pull request by providing the necessary information in the request body. You can update the title, description, and even close the pull request.
```
Comment (tbantle22) [inline content/products/dolthub/api/database.md]:
```suggestion
This API allows you to update a pull request by providing the fields you want to update in the request body. You can update the title, description, and state (only closing a pull request is supported).
```

**Review by tbantle22 — COMMENTED** [review]
Some copy edits and questions. Overall looks good

File: content/.gitbook/assets/updatepull.json (hunk 5 above)
Comment (liuliu-dev) [reply content/.gitbook/assets/updatepull.json]: not all needed, changed the description.

File: content/.gitbook/assets/updatepull.json (hunk 8 above)
Comment (liuliu-dev) [reply content/.gitbook/assets/updatepull.json]: good catch, we can not update this field, removed it, will also remove it from api response.

**Review by tbantle22 — APPROVED** [review]

**Pull request merged**